src/Transfer.js

```diff
@@ -0,0 +1,390 @@
+    labelRemoveIndivitual,
+    leftFooter,
+    leftHeader,
+    maxSelections,
+    optionsWidth,
+    rightFooter,
+    selected,
+    selectedWidth,
+    showFilterInput,
+    onFilterChange,
+}) => {
+    /**
+     * As this component will filter and sort children
+     * in the selected area, adding keys will prevent
+     * reconciliation from suffering badly
+     */
+    const allOptions = addKeyToReactOptions(children)
+
+    /**
+     * Used in the "Filter" section and for
+     * limiting the selectable source options
+     *
+     * Filter can be controlled & uncontrolled.
+     * Providing the "onFilterChange" callback will make it a controlled value
+     */
+    const isFilterEnabled = showFilterInput || onFilterChange
+    const [internalFilter, setInternalFilter] = useState(initialFilter)
+    const actualFilter = onFilterChange ? filter : internalFilter
+
+    /**
+     * These are all the available option react elements.
+     * It will replace all selected options with null
+     */
+    const sourceOptions = filterOutOptions(allOptions, selected)
+    const filteredSourceOptions = getSubsetByFilter({
+        reactOptions: sourceOptions,
+        filter: actualFilter,
+        isFilterEnabled,
+        filterOptions,
+    })
+
+    /**
+     * Extract the selected options. This way custom options are supported
+     * without having to provide a component via the props
+     *
+     * Children are sorted by the order given in the "selected" array.
+     * This is done in order to cover the "append newly selected items
+     * at the end" feature/behavior.
+     */
+    const pickedOptions = extractPickedReactOptions({
+        reactOptions: allOptions,
+        selectedPlainOptions: selected,
+    })
+
+    /**
+     * These are all the highlighted option react elements on the options side.
+     */
+    const {
+        highlightedOptions: highlightedSourceOptions,
+        setHighlightedOptions: setHighlightedSourceOptions,
+        toggleHighlightedOption: toggleHighlightedSourceOption,
+    } = useHighlightedOptions({
+        availableReactOptions: filteredSourceOptions,
+        disabled,
+        maxSelections,
+    })
+
+    /**
+     * These are all the highlighted option react elements on the selected side.
+     */
+    const {
+        highlightedOptions: highlightedPickedOptions,
+        setHighlightedOptions: setHighlightedPickedOptions,
+        toggleHighlightedOption: toggleHighlightedPickedOption,
+    } = useHighlightedOptions({
+        availableReactOptions: pickedOptions,
+        disabled,
+        maxSelections,
+    })
+
+    const {
+        selectSingleOption,
+        deselectSingleOption,
+    } = createDoubleClickHandlers({
+        selectedPlainOptions: selected,
+        setHighlightedSourceOptions,
+        setHighlightedPickedOptions,
+        onChange,
+        maxSelections,
+    })
+
+    return (
+        <Container dataTest={dataTest} className={className} height={height}>
```

Comment:
Minor detail, I see we're passing the base dataTest prop here, and then appending the suffix in the components themselves. The pattern we use in other components is the opposite way (append suffix in parent instead of child). Might be inconsequential, but it is a difference.

Reply:
Done